test/scenarios/applicationmonitoring/without_csi.go

```diff
@@ -0,0 +1,139 @@
+
+	operatorNamespaceBuilder := namespace.NewBuilder(appOnlyDynakube.Namespace)
+
+	assess.InstallOperatorFromSourceWithCustomNamespace(builder, operatorNamespaceBuilder.Build(), appOnlyDynakube)
+
+	assess.InstallDynakubeWithTeardown(builder, &secretConfig, appOnlyDynakube)
+	assess.InstallDynakubeWithTeardown(builder, &secretConfig, appDynakube)
+	builder.Assess("install sample app", sampleApp.Install())
+
+	builder.Assess("create additional pod without ownerreference and check injection", createAdditionalPodToCheckInjection())
+	builder.Assess("create pod with already injected annotation and check injection", checkAlreadyInjected())
+	builder.Assess("create pod with random user and check injection", checkInjectionWithRandomUser())
+
+	builder.Teardown(sampleApp.UninstallNamespace())
+	teardown.UninstallOperatorFromSource(builder, appOnlyDynakube)
+	return builder.Feature()
+}
+
+func createAdditionalPodToCheckInjection() features.Func {
+	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+		resource := environmentConfig.Client().Resources()
+		samplePod := manifests.ObjectFromFile[*corev1.Pod](t, path.Join(project.TestDataDir(), "sample-app/pod-base.yaml"))
+		samplePod.Namespace = sampleAppNamespace
+		require.NoError(t, resource.Create(ctx, samplePod))
+
+		require.NoError(t, resource.Get(ctx, "php-sample", sampleAppNamespace, samplePod))
+
+		require.NotNil(t, samplePod.Spec.InitContainers)
+
+		for _, initContainer := range samplePod.Spec.InitContainers {
+			require.Equal(t, "install-oneagent", initContainer.Name)
+		}
+		return ctx
+	}
+}
+
+func checkAlreadyInjected() features.Func {
+	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+		resource := environmentConfig.Client().Resources()
+		samplePod := manifests.ObjectFromFile[*corev1.Pod](t, path.Join(project.TestDataDir(), "sample-app/pod-base.yaml"))
+		samplePod.Name = "php-sample-already-injected"
+		samplePod.Namespace = sampleAppNamespace
+		alreadyInjected := map[string]string{"oneagent.dynatrace.com/injected": "true"}
+		samplePod.Annotations = alreadyInjected
+		require.NoError(t, resource.Create(ctx, samplePod))
+
+		require.NoError(t, resource.Get(ctx, "php-sample-already-injected", sampleAppNamespace, samplePod))
+
+		require.Nil(t, samplePod.Spec.InitContainers)
+
+		return ctx
+	}
+}
+
+func checkInjectionWithRandomUser() features.Func {
+	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+		resource := environmentConfig.Client().Resources()
+		samplePod := manifests.ObjectFromFile[*corev1.Pod](t, path.Join(project.TestDataDir(), "sample-app/pod-base.yaml"))
+		samplePod.Name = "php-sample-random-user"
+		samplePod.Namespace = sampleAppNamespace
+		samplePod.Spec.SecurityContext.RunAsGroup = address.Of[int64](1234)
+		samplePod.Spec.SecurityContext.RunAsUser = address.Of[int64](1234)
```

Comment:
you could update `test/helpers/sampleapps/sampleapp.go` to have `WithSecurityContext` so you could do something similar to what I mentioned in previous comments